### ISSUE TITLE
php@7.3-mongodb-1.9.1 rebuild needed

### ISSUE THREAD

**Issue by glensc** [kabel/homebrew-pecl#25]
```
➔ brew reinstall php@7.3-mongodb
==> Downloading https://kabel.jfrog.io/artifactory/bottles-pecl/php%407.3-mongodb-1.9.1.big_sur.bottle.tar.gz
Already downloaded: /Users/glen/Library/Caches/Homebrew/downloads/0da9ee49ce6f44dd5c2779d5f403ae863a77f0d076010c26eb17013bcdd0d957--php
==> Reinstalling kabel/pecl/php@7.3-mongodb
==> Pouring php
==> Caveats
php@7.3-mongodb is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

For compilers to find php@7.3-mongodb you may need to set:
  export LDFLAGS="-L/usr/local/opt/php@7.3-mongodb/lib"
  export CPPFLAGS="-I/usr/local/opt/php@7.3-mongodb/include"

==> Summary
🍺  /usr/local/Cellar/php@7.3-mongodb/1.9.1: 245 files, 2.2MB
/usr/local/Homebrew/Library/Taps/glensc/homebrew-tap/lib/php_pecl_formula.rb:61: warning: already initialized constant Class::NAME_PATTERN
/usr/local/Homebrew/Library/Taps/kabel/homebrew-pecl/lib/php_pecl_formula.rb:61: warning: previous definition of NAME_PATTERN was here

$ php -m
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/opt/php@7.3-mongodb/lib/php/20180731/mongodb.so' (tried: /usr/local/opt/php@7.3-mongodb/lib/php/20180731/mongodb.so (dlopen(/usr/local/opt/php@7.3-mongodb/lib/php/20180731/mongodb.so, 9): Library not loaded: /usr/local/opt/icu4c/lib/libicuuc.68.dylib
  Referenced from: /usr/local/opt/php@7.3-mongodb/lib/php/20180731/mongodb.so
```

cc @kabel

**Comment by glensc** [comment]
@kabel have you thought of automating recovering this? like CI cron job that nightly rebuilds if dependency error like this happens?

most of the packages that are broken happen these two libs:
- icu
- imagemagick

**Comment by glensc** [comment]
btw, when rebuilding locally from source, I noticed `chdir` warning. it's harmless, but still there:

```
➔ brew reinstall php@7.3-mongodb -s
==> Downloading https://pecl.php.net/get/mongodb-1.9.1.tgz
######################################################################## 100.0%
==> Reinstalling kabel/pecl/php@7.3-mongodb
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:128: warning: conflicting chdir during another chdir block
==> /usr/local/opt/php@7.3/bin/phpize
==> ./configure --with-php-config=/usr/local/opt/php@7.3/bin/php-config
^C
```

**Comment by glensc** [comment]
actually, it might not be harmless, as source rebuild failed. maybe due some other error.

```
$ brew reinstall php@7.3-mongodb -s
==> Downloading https://pecl.php.net/get/mongodb-1.9.1.tgz
Already downloaded: /Users/glen/Library/Caches/Homebrew/downloads/5627e0b6ad7fe3c715623141c23c9480e735a359729e26945f0072e4cd229ebc--mongodb-1.9.1.tgz
==> Reinstalling kabel/pecl/php@7.3-mongodb
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:128: warning: conflicting chdir during another chdir block
==> /usr/local/opt/php@7.3/bin/phpize
==> ./configure --with-php-config=/usr/local/opt/php@7.3/bin/php-config
==> make
Error: File exists @ dir_s_mkdir - /usr/local/Cellar/php@7.3-mongodb/1.9.1/.brew
```

**Comment by kabel** [comment]
Yes, those upstream libs are known to be problems because every update is like a breaking change for no good reason.
No, I have not considered automating bumps for them.

**Comment by glensc** [comment]
Yep they need to learn about soname versioning. But instead, they took easy way out, changing soname with every release.

Actually I think ImageMagick has gone better, they don't bump soname with every release now.

BTW, I think pecl imagick can be built with graphicsmagick, as that has less releases and proper soname versioning.

So it would solve that problem, but it likely lags behind with features, so that's not probably expected by pecl package users.

**Comment by kabel** [comment]
Fixed at 464b05cb71dde3b0d15c0cb2ea529fe729c384d4.
